compiler/parser.nim

```diff
@@ -2203,7 +2174,7 @@ proc complexOrSimpleStmt(p: var Parser): PNode =
   #|                     | bindStmt | mixinStmt)
   #|                     / simpleStmt
   case p.tok.tokType
-  of tkIf: result = parseIfOrWhen(p, nkIfStmt)
+  of tkIf: result = if maybeExpr: parseIfExpr(p, nkIfStmt) else: parseIfOrWhen(p, nkIfStmt)
```

Comment:
This line means you implemented an "attribute grammar" but Nim's grammar does not reflect that.

Reply:
I can move this to `semiStmtList` if you want to, since it is the only part where its actually used and `semiStmtList` seems to lack a representation in the grammar anyways.

Reply:
Maybe I should add this:
```

  #| complexOrSimpleStmt(maybeExpr = true) = (ifExpr | ifStmt | whenStmt | whileStmt
  #|                     | tryStmt | forStmt
  #|                     | blockStmt | staticStmt | deferStmt | asmStmt
  #|                     | 'proc' routine
  #|                     | 'method' routine
  #|                     | 'func' routine
  #|                     | 'iterator' routine
  #|                     | 'macro' routine
  #|                     | 'template' routine
  #|                     | 'converter' routine
  #|                     | 'type' section(typeDef)
  #|                     | 'const' section(constant)
  #|                     | ('let' | 'var' | 'using') section(variable)
  #|                     | bindStmt | mixinStmt)
  #|                     / simpleStmt
```
to the grammar comment?

Reply:
That's not how attributed grammars work... ;-)

Reply:
Moved the condition out of `complexOrSimpleStmt` and update the grammar accordingly.